src/app/Shared/MatchExpressionEvaluator.tsx

```diff
@@ -61,37 +59,34 @@ import {
   InfoCircleIcon,
   WarningTriangleIcon,
 } from '@patternfly/react-icons';
+import _ from 'lodash';
 import * as React from 'react';
 
 export interface MatchExpressionEvaluatorProps {
   inlineHint?: boolean;
   matchExpression?: string;
   onChange?: (validated: ValidatedOptions) => void;
+  onTargetChange?: (target: Target) => void;
 }
 
 export const MatchExpressionEvaluator: React.FC<MatchExpressionEvaluatorProps> = ({
   inlineHint,
   matchExpression,
   onChange,
+  onTargetChange,
 }) => {
-  const context = React.useContext(ServiceContext);
-  const addSubscription = useSubscriptions();
-  const [target, setTarget] = React.useState(undefined as Target | undefined);
+  const [target, setTarget] = React.useState(NO_TARGET);
   const [valid, setValid] = React.useState(ValidatedOptions.default);
   const [copied, setCopied] = React.useState(false);
 
   React.useEffect(() => {
-    addSubscription(context.target.target().subscribe(setTarget));
-  }, [addSubscription, context.target, setTarget]);
-
-  React.useEffect(() => {
-    if (!matchExpression || !target?.connectUrl) {
+    if (!matchExpression || !target.connectUrl) {
       setValid(ValidatedOptions.default);
       return;
     }
     try {
       const f = new Function('target', `return ${matchExpression}`);
-      const res = f(target);
+      const res = f(_.cloneDeep(target));
```

Comment:
Probably worth extracting this to a utility function since it's used in two places now.